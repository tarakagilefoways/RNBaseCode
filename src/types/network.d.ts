export interface ApiResponseType<T> {
  status: boolean;
  message: string;
  data: T;
}
