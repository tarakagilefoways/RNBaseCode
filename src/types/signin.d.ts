interface ILogiInReq {
  username: string;
  password: string;
}
interface ILogInRes {
  status: number;
  message: string;
  data: AuthData;
}
