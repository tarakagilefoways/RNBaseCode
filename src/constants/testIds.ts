export const SignInTIDS = {
  Username: 'Username',
  UsernameError: 'UsernameError',
  Password: 'Password',
  PasswordError: 'PasswordError',
  SignIn: 'Signin',
  LoginText: 'LoginText',
  NetworkError: 'NetworkError',
};
