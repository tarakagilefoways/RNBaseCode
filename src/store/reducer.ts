import { combineReducers } from '@reduxjs/toolkit';
import authReducer from '../service/auth/slice';

const rootReducer = combineReducers({
  auth: authReducer,
});
export default rootReducer;
