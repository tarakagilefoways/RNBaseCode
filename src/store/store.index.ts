import {combineReducers, configureStore} from "@reduxjs/toolkit";
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from "redux-persist";
import createSensitiveStorage from "redux-persist-sensitive-storage";

import AuthReducer from "../service/auth/slice";

const rootReducer = combineReducers({
  auth: AuthReducer.reducer,
});
const storage = createSensitiveStorage({
  keychainService: "rrr",
  sharedPreferencesName: "rrr",
});
const config = {
  key: "root",
  storage,
};
const reducer = persistReducer(config, rootReducer);

export const store = configureStore({
  reducer: reducer,
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});
export const persistor = persistStore(store);
