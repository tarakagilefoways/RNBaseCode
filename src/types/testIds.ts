export const SignInTIDS = {
  Email: 'Email',
  EmailError: 'EmailError',
  Password: 'Password',
  PasswordError: 'PasswordError',
  SignIn: 'Signin',
  LoginText: 'LoginText',
};
