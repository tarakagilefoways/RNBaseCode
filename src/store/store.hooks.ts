import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import RootReducer from './reducer';
import store from './store.index';

export type RootReducerType = ReturnType<typeof RootReducer>;
export type RootStoreType = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;
export type AppSelector = typeof store.getState;

export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootStoreType> = useSelector;
