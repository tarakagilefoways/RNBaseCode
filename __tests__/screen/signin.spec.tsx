import React from 'react';
import renderer from 'react-test-renderer';
import { RenderResult, act, fireEvent } from '@testing-library/react-native';

import { SignInTIDS } from '../../src/types/testIds';
// import App from '../../App';
import AppStack, { MainStackParamList } from '../../src/router/root.index';
import renderWithProviders from '../../src/utils/test-utils';
import { NavigationContainer } from '@react-navigation/native';

import Home from '../../src/screen/home';
import SignInForm from '../../src/screen/SignIn';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { RootReducerType } from '../../src/store/store.hooks';
const Stack = createNativeStackNavigator<MainStackParamList>();

describe('Sign In Form Snapshot', () => {
  test('Renders Correctly and Create Snapshot', () => {
    const tree = renderWithProviders(
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="SignInForm" component={SignInForm} />
          <Stack.Screen name="Home" component={Home} />
        </Stack.Navigator>
      </NavigationContainer>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Sign In Form Test Suit:', () => {
  let wrapper: RenderResult;

  let email = 'Test@Email.com';
  let password = 'Test@123';

  beforeEach(() => {
    wrapper = renderWithProviders(
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="SignInForm" component={SignInForm} />
          <Stack.Screen name="Home" component={Home} />
        </Stack.Navigator>
      </NavigationContainer>,
    );
  });

  it('Check Input Components Exists', () => {
    let emailInput = wrapper.getByTestId(SignInTIDS.Email);
    expect(emailInput).toBeTruthy();
    let passwordInput = wrapper.getByTestId(SignInTIDS.Password);
    expect(passwordInput).toBeTruthy();
  });

  it('Get Button  Components exist', () => {
    let signInBtn = wrapper.getByTestId(SignInTIDS.SignIn);
    expect(signInBtn).toBeTruthy();
  });

  it('Update Input Components Text and Check Later', async () => {
    let emailInput = wrapper.getByTestId(SignInTIDS.Email);
    let passwordInput = wrapper.getByTestId(SignInTIDS.Password);

    await act(() => {
      fireEvent.changeText(emailInput, email);
      fireEvent.changeText(passwordInput, password);
    });

    let updatedEmailInput = wrapper.getByDisplayValue(email);
    expect(updatedEmailInput).toBeTruthy();
    let updatedPasswordInput = wrapper.getByDisplayValue(password);
    expect(updatedPasswordInput).toBeTruthy();
  });

  it('Presse Sign In and Check Screen with Login Test', async () => {
    let emailInput = wrapper.getByTestId(SignInTIDS.Email);
    let passwordInput = wrapper.getByTestId(SignInTIDS.Password);
    let signInBtn = wrapper.getByTestId(SignInTIDS.SignIn);

    await act(() => {
      fireEvent.changeText(emailInput, email);
      fireEvent.changeText(passwordInput, password);
      fireEvent.press(signInBtn);
    });

    let textHomeScreen = wrapper.getByText('Home Screen');
    expect(textHomeScreen).toBeTruthy();
  });
});

describe('Sign In Form Prefilled Test Suit:', () => {
  let wrapper: RenderResult;

  let email = 'TestPrefill@Email.com';
  let password = 'TestPrefill@123';
  const initialState: RootReducerType = {
    auth: { email: email, password: password, isLoading: false },
  };
  beforeEach(() => {
    wrapper = renderWithProviders(
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="SignInForm" component={SignInForm} />
          <Stack.Screen name="Home" component={Home} />
        </Stack.Navigator>
      </NavigationContainer>,
      { preloadedState: initialState },
    );
  });

  it('Presse Sign In and Check Screen with Login Test', async () => {
    let signInBtn = wrapper.getByTestId(SignInTIDS.SignIn);
    await act(() => {
      fireEvent.press(signInBtn);
    });

    let textHomeScreen = wrapper.getByText('Home Screen');
    expect(textHomeScreen).toBeTruthy();
  });
});
