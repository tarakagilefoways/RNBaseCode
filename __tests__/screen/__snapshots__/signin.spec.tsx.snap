// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In Form Snapshot Renders Correctly and Create Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="SignInForm"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
                "paddingVertical": 16,
              }
            }
          >
            <TextInput
              onChangeText={[Function]}
              placeholder="Email"
              style={
                {
                  "alignSelf": "center",
                  "borderWidth": 0.5,
                  "fontSize": 18,
                  "height": 40,
                  "marginTop": 8,
                  "padding": 4,
                  "width": "80%",
                }
              }
              testID="Email"
              value=""
            />
            <Text
              style={
                {
                  "alignSelf": "center",
                  "fontSize": 18,
                  "marginTop": 8,
                  "padding": 4,
                  "width": "80%",
                }
              }
              testID="EmailError"
            />
            <TextInput
              onChangeText={[Function]}
              placeholder="Password"
              secureTextEntry={true}
              style={
                {
                  "alignSelf": "center",
                  "borderWidth": 0.5,
                  "fontSize": 18,
                  "height": 40,
                  "marginTop": 8,
                  "padding": 4,
                  "width": "80%",
                }
              }
              testID="Password"
              value=""
            />
            <Text
              style={
                {
                  "alignSelf": "center",
                  "fontSize": 18,
                  "marginTop": 8,
                  "padding": 4,
                  "width": "80%",
                }
              }
              testID="PasswordError"
            />
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
              testID="Signin"
            >
              <View
                style={
                  [
                    {},
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#007AFF",
                        "fontSize": 18,
                        "margin": 8,
                        "textAlign": "center",
                      },
                    ]
                  }
                >
                  Sign In
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
